File: pa3/pa3 w: absolute paths/rmi/Beacon.java
import java.util.Random;
import java.io.Serializable;
import java.rmi.RemoteException;

/*
	@Author Josh Loftus
	Beacon object that is sent between the Beacon Sender and the Beacon Listener Register using rmi
*/
public class Beacon implements Serializable {

	int id;
	long start_up_time;
	int time_interval;
	static final long serialVersionUID = 0;

	public Beacon(CmdRegister cmd_register) {
		Random random = new Random();
		this.id = random.nextInt(1000) + 1;
		GetLocalTime glt = new GetLocalTime();
		try {
			glt = (GetLocalTime) cmd_register.cmd_service.execute("GetLocalTime", glt);
		} catch(RemoteException e) { e.printStackTrace(); }
		this.start_up_time = glt.time;
		this.time_interval = Agent.TIME_INTERVAL;
	}
}

File: pa1/agent/Beacon.cpp
// class not created by me!
class Beacon {
	public:
		int id;
		int start_up_time;
		char* ip;
		int time_interval;
		int cmd_port;

		// beacon object outlined in the hw
		Beacon() {}

		Beacon(int _id, int _start_up_time, char* _ip, int _time_interval, int _cmd_port) {
			id = _id;
			start_up_time = _start_up_time;
			ip = _ip;
			time_interval = _time_interval;
			cmd_port = _cmd_port;
		}
};